### DISCUSSION TITLE
How to append command with parameters?

### DISCUSSION THREAD

**Discussion by mko543** [westacks/telebot#68] [Q&A] [answered]
How to make command with parameters?
```php
class GenerateSubKey extends BaseCommands
{
    protected static $aliases = [ '/generate', '/GetKey' ];
    protected static $description = 'Send "/generate" or "/GetKey" to get subscription key';

    public function handle()
    {
        if ($this->getClient()->client_id == config('settings.admin_id')) {
            // Generate a subscription key
            $generated_key = Str::uuid()->toString();
            $days=30;
            $expire = Carbon::now()->addDays($days);

            // save it
            $subscription = $this->CreateKey($generated_key,$expire);

            if ($subscription) {
                // send it to the admin
                $this->sendMessage(["text" => $generated_key]);
            } else {
                $this->sendMessage(["text" => 'Something Wrong!.']);
            }
        }
    }
    protected function CreateKey($generated_key,$expire){
        $subscription = new Subscription();
            $subscription->key = $generated_key;
            $subscription->expire_at = $expire;
            $result = $subscription->save();
            return $result;
    }
}
```
I want to make  $days parmater in command that i can type : `/generate <number of days>`

**Comment by punyflash** [comment, marked as answer]
You can create a method to extract arguments from your command:
```php
protected function arguments()
{
    $command = array_filter(explode(' ', $this->update->message()->text));
    array_shift($command);
    return array_values($command);
}
```

This is the simplest way, but If you need more complex arguments, you may add a custom algorithm using regex or other.

Then just use this method (for example, if command was `/generate 5`):
```php
[$days] = $this->arguments();
// $days == 5
```

**Reply by mko543** [reply]
If there is no method to do it by the SDK, Then i think this a prefect answer 👌

Thank you for help

**Reply by mko543** [reply]
> You can create a method to extract arguments from your command:
> 
> ```
> protected function arguments()
> {
>     $command = array_filter(explode(' ', $this->update->message()->text));
>     array_shift($command);
>     return array_values($command);
> }
> ```
> 
> This is the simplest way, but If you need more complex arguments, you may add a custom algorithm using regex or other.
> 
> Then just use this method (for example, if command was `/generate 5`):
> 
> ```
> [$days] = $this->arguments();
> // $days == 5
> ```

BTW for anyone will use this if get any error just reset the bot.
